core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -621,14 +621,22 @@ object SparkSubmit {
     if (isKubernetesCluster) {
       childMainClass = "org.apache.spark.deploy.kubernetes.submit.Client"
       if (args.isPython) {
+        childArgs += "--py-file"
         childArgs += args.primaryResource
+        childArgs += "--main-class"
         childArgs += "org.apache.spark.deploy.PythonRunner"
+        childArgs += "--other-py-files"
         childArgs += args.pyFiles
       } else {
+        childArgs += "--primary-java-resource"
```

Comment:
In general I'm against having logic that requires file names to have a certain extension.